src/Graph.ts

```diff
@@ -453,7 +446,16 @@ export default class Graph {
 
 			module.dependencies.forEach(depModule => {
 				if (!depModule.isExternal) {
-					visit(<Module>depModule, seen);
+					if (depModule.id in parents) {
+						if (!allSeen[depModule.id]) {
+							parents[depModule.id] = module.id;
+							cycles.push({ id: depModule.id, parents: { ...parents } });
```

Comment:
Could we not just output the warning directly here instead of collecting the cycles list?

Reply:
Sure, I just did that.